File: solutions/javascript/resistor-color-duo/1/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const decodedValue = (color) => {
  const ditgit1 = COLORS.indexOf(color[0]);
  const ditgit2 = COLORS.indexOf(color[1]);
  const result = ditgit1 * 10 + ditgit2;
  return result;
};

export const COLORS = [
  "black", "brown", "red", "orange", "yellow",
  "green", "blue", "violet", "grey", "white"
];

File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
class ResistorColorDuo {
    
    private final String[] colorList = {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    };
    
    int value(String[] colors) {
        int first = colorCode(colors[0]);
        int second = colorCode(colors[1]);

        return first * 10 + second;
    }

    private int colorCode(String color) {
        for(int i = 0; i < colorList.length; i++) {
            if(colorList[i].equals(color)) {
                return i;
            }
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }
}
